src/darwin/Framework/CHIP/NSSetCATConversion.h

```diff
@@ -0,0 +1,74 @@
+/**
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#import <Foundation/Foundation.h>
+
+#import "MTRLogging_Internal.h"
+
+#include <lib/core/CASEAuthTag.h>
+#include <lib/core/CHIPError.h>
+#include <lib/support/SafeInt.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * Utilities for converting between NSSet<NSNumber *> and chip::CATValues.
+ */
+
+inline CHIP_ERROR ToCATValues(NSSet<NSNumber *> * catSet, chip::CATValues & values)
```

Comment:
Why inline? These seem to have a substantial amount of code in them so might be better as a regular function.

Reply:
It was mostly modeled on the existing string/data conversions, but you are right.  It would be better to have these out-of-line.

Reply:
@ksperling-apple Thoughts on the file name while we are here?  I modeled this on NSDataSpanConversion.h and NSStringSpanConversion.h, but maybe I should name this MTRSomething instead...

Reply:
Per Slack discussion:

1. Put the declarations in MTRConversion.h
2. Put the impls in a (new) MTRConversion.mm

Reply:
At some point we could just move those NSData / NSString conversion helpers into MTRConversion.h too.